lib/rules/no-constant-condition.js

```diff
@@ -32,11 +32,15 @@ module.exports = function(context) {
                 return true;
 
             case "UnaryExpression":
-                return isConstant(node.argument);
+                return node.operator === "typeof" || isConstant(node.argument);
 
             case "BinaryExpression":
             case "LogicalExpression":
-                return isConstant(node.left) && isConstant(node.right) && node.operator !== "in";
+                return node.left.operator !== "typeof" &&
```

Comment:
I am able to get only two options to this . Here is my thought process . 
In case of LogicalExpression such as `abc === 'str'`  , we are checking `isConstant(abc) && isConstant('str')` . The first one gives false and second true , so over all false . We are safe. 

In case of LogicalExpression with `typeof`  such as `typeof x === 'str'` , `isConstant(typeof x) && isConstant('str')`. Right naturally gives true, and left also gives true because we explicitly told to consider that as a constant condition. So it returns true which is not what we want . So , I am saying if you find typeof operator , dont bother checking isConstant() is true , just consider the overall condition is not constant . I think for constant condition validation this should be sufficient .

The other possibility is to pass a param to `isConstant` something like: 
```js
function isConstant(node, isUnary = true ){
...
return ( node.operator == 'typeof' && isUnary )
```

and pass `isUnary` from LogiclExpression as false . 

 I am open to learn and fine with any corrections or suggestions , please help me in getting best code .